src/scripts/search.js

```diff
@@ -0,0 +1,135 @@
+
+if (window.location.href.includes('/fr/')) {
+  lang = 'fr';
+}
+
+if (!index) {
+  index = 1;
+}
+
+if (searchTerm) {
+  // I realized we don't have a value property for gcds-search, want to add that
+  // document.getElementById('searchbar').value = searchTerm;
+  const search = await pagefind.search(searchTerm);
+  const results = search.results;
+
+  const langObject = {
+    en: {
+      results: `${results.length} search results for "${searchTerm}"`,
+      noresults: 'No results',
+      noresultsbody: `
+        <p class="mb-400">No pages were found that match your search terms.</p>
+
+        <p class="mb-400">Suggestions:</p>
+
+        <ul class="list-disc mb-400">
+          <li>Make sure all search terms are spelled correctly</li>
+          <li>Try different search terms</li>
+          <li>Try more general search terms</li>
+        </ul>
+      `,
+    },
+      
+    fr: {
+      results: `${results.length} résultats de recherche pour « ${searchTerm} »`,
+      noresults: 'Aucun résultat',
+      noresultsbody: `
+        <p class="mb-400">Aucun résultat ne correspond à vos critères de recherche.</p>
+
+        <p class="mb-400">Suggestions:</p>
+
+        <ul class="list-disc mb-400">
+          <li>Assurez-vous que tous vos termes de recherches sont bien orthographiés</li>
+          <li>Utilisez de différents termes de recherche</li>
+          <li>Utilisez des termes de recherche plus généraux</li>
+        </ul>
+      `,
+    }
+  }
+
+  // Results length
+  if (results.length > 0) {
+    // Results heading
+    let resultsHeading = document.createElement('gcds-heading');
+    resultsHeading.setAttribute('tag', 'h2');
+    resultsHeading.innerHTML = langObject[lang].results;
+    document.getElementById('results-count').append(resultsHeading);
+
+    const totalPages = Math.ceil(results.length / 10);
+
+    let pageResults = results.slice(10 * (index - 1), 10 * index + 1);
+
+    // length could also be from search.unfilteredResultCount
+    for (const result of pageResults) {
+      let data = await result.data();
+      addResult(data);
+    }
+
+    if (totalPages > 1) {
+      const pagination = document.createElement('gcds-pagination');
+      pagination.display = 'list';
+      pagination.currentPage = index;
+      pagination.totalPages = totalPages;
+      pagination.url = {
+        queryStrings: {
+          'q': searchTerm,
+          'page::match': '{{1}}',
+        },
+      };
+      pagination.classList.add('mt-400');
+      document.getElementById('pagination').appendChild(pagination);
+    }
+  } else if (searchTerm && results.length === 0) {
+    let resultsHeading = document.createElement('gcds-heading');
+    resultsHeading.setAttribute('tag', 'h2');
+    resultsHeading.innerHTML = langObject[lang].noresults;
+    document.getElementById('results-count').append(resultsHeading);
+
+    document.getElementById('results').innerHTML = langObject[lang].noresultsbody;
+  }
+}
+
+function addResult(result) {
```

Comment:
This function currently does two things: creating DOM elements and appending them to the document. Could we split them up into two functions to add some separation of concerns? One for creating elements and another for appending them